core/src/test/java/org/elasticsearch/action/support/master/IndexingMasterFailoverIT.java

```diff
@@ -0,0 +1,119 @@
+package org.elasticsearch.action.support.master;
+
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.discovery.DiscoverySettings;
+import org.elasticsearch.discovery.zen.elect.ElectMasterService;
+import org.elasticsearch.discovery.zen.fd.FaultDetection;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.disruption.NetworkDisconnectPartition;
+import org.elasticsearch.test.disruption.NetworkPartition;
+import org.elasticsearch.test.transport.MockTransportService;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+@ESIntegTestCase.SuppressLocalMode
```

Comment:
If I remove SuppressLocalMode, and add ("discovery.type", "zen") to the test, I hit the issue with zendiscovery + local mode again:

```
[2015-11-04 14:38:31,931][WARN ][org.elasticsearch.discovery] [node_t0] waited for 30s and no initial state was set by the discovery
[2015-11-04 14:38:31,932][INFO ][org.elasticsearch.node   ] [node_t0] started
[2015-11-04 14:38:32,042][WARN ][org.elasticsearch.discovery] [node_t1] waited for 30s and no initial state was set by the discovery
[2015-11-04 14:38:32,043][INFO ][org.elasticsearch.node   ] [node_t1] started
[2015-11-04 14:38:32,146][WARN ][org.elasticsearch.discovery] [node_t2] waited for 30s and no initial state was set by the discovery
```

and the following exception at the end:

```
MasterNotDiscoveredException[]
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:167)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.start(TransportMasterNodeAction.java:123)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:81)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:52)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:70)
	at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:47)
	at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:43)
	at org.elasticsearch.transport.local.LocalTransport.handleRequest(LocalTransport.java:290)
	at org.elasticsearch.transport.local.LocalTransport.messageReceived(LocalTransport.java:251)
	at org.elasticsearch.transport.local.LocalTransport$1.run(LocalTransport.java:229)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

I am still investigating that one...